ibis/expr/types/relations.py

```diff
@@ -74,14 +75,7 @@ def f(
 
 
 @public
-class TableExpr(Expr):
-    @property
-    def _factory(self):
-        def factory(arg):
-            return TableExpr(arg)
-
-        return factory
-
+class TableExpr(Expr, HasName):
```

Comment:
What does `TableExpr.name` compile to? Something like `(SELECT * FROM t) name`?

Reply:
Bump

Reply:
What do you mean under compile to? `TableExpr.name` is a method which instantiates the `ops.Alias()` objects.

Reply:
Right, this PR introduces a `name` on `TableExpr`s that doesn't currently exist. What code does the new `ops.Alias` translate to?

Reply:
Now I managed to understand :)

Indeed name() hasn't been exposed on TableExpr before, so I removed the HasName trait and defined the has_name(), get_name() methods on Expr base class and added name() method to ValueExpr.